### PR TITLE
Add ability to remove RE-matched lines before diff.

### PR DESCRIPTION
**Pull request by zzehring**
This change adds a '--remove-matching-lines' flag which takes in a
RegExp string. When specified, the tool will first remove any lines in
the source files that match the RegExp and then take the diff. This
way, we can do some preprocessing before we take the diff.

### CODE REVIEW

**Comment by zzehring** [comment]
This feature is useful because there are scenarios where we want to remove some attributes that we don't want to show up in the diff (e.g. we want the diff, but we don't care about the flux annotations/labels that exist in the output). 

We could attempt to do this by passing through the `--ignore-matching-lines=RE` (i.e. Ignore changes whose lines all match RE) in the diff command itself, however this is not quite as effective [due to a caveat of `diff -I`](https://www.gnu.org/software/diffutils/manual/html_node/Specified-Lines.html#Specified-Lines). 

TL;DR:
> for each nonignorable change, diff prints the complete set of changes in its vicinity, including the ignorable ones.

So, often in the case of comparing in a flux environment, we'd always get some diff noise when checking the diff of a flux managed resource.
